test/unit-tests/sexp.mlt

```diff
@@ -140,7 +141,12 @@ parse {|"$bar%foo%"|}
 
 parse {|\%{foo}|}
 [%%expect{|
-- : parse_result = Same (Ok [\%{foo}])
+- : parse_result =
+Different
+ {jbuild =
+   Ok
+    [<printer pp_sexp_ast raised an exception: Invalid_argument("atom '\\%{foo}' cannot be in dune syntax")>];
```

Comment:
Maybe we should do `#remove_printer pp_sexp_ast` before these tests, it took me a while to understand what was going on here

Reply:
Hmm, I think we should print the atoms unambigiously instead. Otherwise it sucks that we can't pp some of these ast's for debugging.